presto-spark-base/src/test/java/com/facebook/presto/spark/execution/http/TestPrestoSparkHttpWorkerClient.java

```diff
@@ -0,0 +1,413 @@
+                                    }
+                                }
+                                // GET /v1/task/{taskId}/results/{bufferId}/{token}
+                                else {
+                                    try {
+                                        future.complete(responseHandler.handle(
+                                                request,
+                                                MockResponse.createResultResponse(
+                                                        HttpStatus.OK,
+                                                        taskId,
+                                                        0,
+                                                        1,
+                                                        true)));
+                                    }
+                                    catch (Exception e) {
+                                        e.printStackTrace();
+                                        future.completeExceptionally(e);
+                                    }
+                                }
+                            }
+                        }
+                        else if (method.equalsIgnoreCase("DELETE")) {
+                            // DELETE /v1/task/{taskId}
+                            try {
+                                future.complete(responseHandler.handle(request,
+                                        MockResponse.createDummyResultResponse()));
+                            }
+                            catch (Exception e) {
+                                e.printStackTrace();
+                                future.completeExceptionally(e);
+                            }
+                        }
+                        future.completeExceptionally(
+                                new Exception("Unknown path " + path));
+                    },
+                    (long) mockDelay.getValue(),
+                    mockDelay.getUnit());
+            return future;
+        }
+
+        @Override
+        public RequestStats getStats()
+        {
+            return null;
+        }
+
+        @Override
+        public long getMaxContentLength()
+        {
+            return 0;
+        }
+
+        @Override
+        public void close() {}
+
+        @Override
+        public boolean isClosed()
+        {
+            return false;
+        }
+
+        private String getTaskId(URI uri)
+        {
+            String fromTaskId = uri.getPath().substring(
+                    TASK_ROOT_PATH.length() + 1);
+            int endPos = fromTaskId.indexOf("/");
+            if (endPos < 0) {
+                return fromTaskId;
+            }
+            return fromTaskId.substring(0, endPos);
+        }
+    }
+
+    public static class MockResponse
+            implements Response
+    {
+        private final int statusCode;
+        private final String statusMessage;
+        private ListMultimap<HeaderName, String> headers;
```

Comment:
- final
- Multimap

Reply:
made it final.
cannot change to Multimap because Response super class exposes the API as ListMultiMap